rust/cloud-storage/email_service/src/util/upload_attachment.rs

```diff
@@ -273,16 +317,15 @@ async fn upload_data_to_presigned_url(
         .body(attachment_data)
         .send()
         .await
-        .context("HTTP PUT request to presigned URL failed")?;
+        .map_err(|e| UploadAttachmentError::PresignedUrlUploadFailed(e.to_string()))?;
 
     if !response.status().is_success() {
         let status = response.status();
         let body = response.text().await.unwrap_or_default();
-        return Err(anyhow!(
-            "Failed to upload attachment to presigned url: {} {}",
-            status,
-            body
-        ));
+        return Err(UploadAttachmentError::PresignedUrlUploadFailed(format!(
```

Comment:
the presignedurluploadfailed error could take the status and body as args then you can format the error message with the derive error

i think it would be cleaner for consistency